braindecode/models/modules.py

```diff
@@ -528,102 +529,289 @@ def extra_repr(self) -> str:
+        a time-domain design technique that generally gives improved
+        attenuation using fewer samples than "firwin2".
+    pad : str, default='reflect_limited'
+        The type of padding to use. Supports all func:`numpy.pad()` mode options.
+        Can also be "reflect_limited", which pads with a reflected version of
+        each vector mirrored on the first and last values of the vector,
+        followed by zeros. Only used for ``method='fir'``.
+    verbose: bool | str | int | None, default=True
+        Control verbosity of the logging output. If ``None``, use the default
+        verbosity level. See the func:`mne.verbose` for details.
+        Should only be passed as a keyword argument.
     """
 
     def __init__(
         self,
-        orig_freq: int = 16000,
-        new_freq: int = 16000,
-        resampling_method: str = "sinc_interp_hann",
-        lowpass_filter_width: int = 6,
-        rolloff: float = 0.99,
-        beta: Optional[float] = None,
-        *,
-        dtype: Optional[torch.dtype] = None,
-    ) -> None:
-        super().__init__()
+        n_chans: int,
+        sfreq: int,
+        band_filters: Optional[List[Tuple[float, float]] | int] = None,
+        filter_length: str | float | int = "auto",
+        l_trans_bandwidth: str | float | int = "auto",
+        h_trans_bandwidth: str | float | int = "auto",
+        method: str = "fir",
+        phase: str = "zero",
+        iir_params: Optional[dict] = None,
+        fir_window: str = "hamming",
+        fir_design: str = "firwin",
+        verbose: bool = True,
+    ):
+        super(FilterBank, self).__init__()
+
+        # The first step here is to check the band_filters
+        # We accept as None values.
+        if band_filters is None:
+            """
+            the filter bank is constructed using 9 filters with non-overlapping
+            frequency bands, each of 4Hz bandwidth, spanning from 4 to 40 Hz
+            (4-8, 8-12, …, 36-40 Hz)
+
+            Based on the reference: FBCNet: A Multi-view Convolutional Neural
+            Network for Brain-Computer Interface
+            """
+            band_filters = [(low, low + 4) for low in range(4, 36 + 1, 4)]
+        # We accept as int.
+        if isinstance(band_filters, int):
+            warn(
+                "Creating the filter banks equally divided in the "
+                "interval 4Hz to 40Hz with almost equal bandwidths. "
+                "If you want a specific interval, "
+                "please specify 'band_filters' as a list of tuples.",
+                UserWarning,
+            )
+            low_freq = 4
+            high_freq = 40
+
+            band_filters = self._create_band_filters(low_freq, high_freq, band_filters)
+        if not isinstance(band_filters, list):
+            ValueError(
+                "`band_filters` should be a list of tuples if you want to "
+                "use them this way."
+            )
```

Comment:
Add check that the `band_filters[0]` is splitable in 2 values.

Reply:
checker made